clode/solver.py

```diff
@@ -34,41 +27,62 @@ class Stepper(Enum):
     stochastic_euler = "seuler"
 
 
-# note: superset of all possible options. not all used in all cases...
-# - e.g., RK4 has no dtmax/abstol/reltol. Feaures doesn't use max_store/nout
-# --> can we expose just the relevant ones, given solver object + stepper?
-class SolverParams:
-    def __init__(
-        self,
-        dt: float = 0.1,
-        dtmax: float = 1.0,
-        abstol: float = 1e-6,
-        reltol: float = 1e-3,
-        max_steps: int = 1000000,
-        max_store: int = 1000000,
-        nout: int = 1,
-    ):
-        self.dt = (dt,)
-        self.dtmax = (dtmax,)
-        self.abstol = (abstol,)
-        self.reltol = (reltol,)
-        self.max_steps = (max_steps,)
-        self.max_store = (max_store,)
-        self.nout = (nout,)
-        self._sp = _clode.solver_params(
-            dt, dtmax, abstol, reltol, max_steps, max_store, nout
-        )
+# base solver class with only transient()
+class Simulator:
+    """Base class for all simulators.
 
+    This class is used to simulate transient behavior of a system of ODEs.
```

Comment:
This description isn't quite accurate - it can't find separatrix solutions on its own (those are very special trajectories, best found via other methods). It could be used to find steady states, as long as that's what you expect from your system - which may not always be the case. Again, there are other specialized methods for doing that directly even for systems that oscillate (essentially, Newton's method or other nonlinear system solvers). 

I'd write:
This class is intended as a base class for other simulators. It provides the core functionality to solve a system of ODEs without storing any intermediate results. It can be used directly if only the final system states are desired.